### ISSUE TITLE
Support @directive(arg: value)

### ISSUE THREAD

**Issue by barbieri** [profusion/sgqlc#6]
Support directives, such as:

```graphql
query aQuery($ignoreX: Boolean) {
  x @skip(if: $ignoreX)
}
```

**Comment by barbieri** [comment]
We do not handle the directives at all and I never stopped to think about what to do about them.

For instance, on the `schema`, we have projects like https://github.com/profusion/apollo-validation-directives/ where we could call local wrappers, like converting/validating types locally before sending the query or receiving the values (similar to the custom scalars).

On the `operation` (executable), we could mark statements with directives, such as `@skip()` above. Then let the server know how to process that information.

But how to integrate it's the cumbersome bit, maybe follow the `.__fields__()` and provide a `.__directives__()` and let the user specify name + args?

**Comment by maaft** [comment]
I think directives are very important. They are used a lot in most schema-first databases like nexus, dgraph, etc..

I'll definitely look how to integrate them (because I need them) but this might be not before may this year.
If you have any more concrete ideas until then, I'm happy to hear about them!

**Comment by barbieri** [comment]
yes, we use a lot in our schema-first servers (that's why we created that apollo-validation-directives repo). But for clients, it's less frequently used, that's why I never bothered.

In the schema I'd use something like the `__directives__()` as I said, can create a `Directive()` that can be inherited, provide the name and arguments using what exists. And when interpreting the results, in addition to the scalar conversion we'd give it to the directive, allowing to do extra validations.

But to do operations, need to see if we could do some `__directives__()` as well.

**Comment by aychang95** [comment]
Hi @barbieri, any updates on supporting @directives for sgqlc?

We've got queries that use @include and support for this would be very helpful!

**Comment by barbieri** [comment]
@aychang95 unfortunately out of time to implement it. Any help is appreciated :-)

In the operations generator, you'd have to search for the fragment casting (`__as__`) or aliasing (`__alias__`) and follow that path.

**Comment by sandaliqbal** [comment]
Can you please suggest how to coden for @include directive? I have a gql something like this:

edges {
      node {
        agents @include(if: $agents) {
          agentName
          firstSeen
          lastSeen
          sourceFirstSeen
          sourceLastSeen
          __typename
        }
but when I try to create code for this I get this error:
no argument named 'if' on field Device.agents 
Any idea on how this could be resolved?

**Comment by barbieri** [comment]
@sandaliqbal not supported and I don't have time to do it myself, patches are welcome.
